src/networking/index.js

```diff
@@ -32,32 +30,32 @@ export default class {
   init(config: Config) {
     this._config = config;
 
-    this._maxSubDomain = 20;
+    this._maxSubDomain = 10;
     this._currentSubDomain = Math.floor(Math.random() * this._maxSubDomain);
-    this._providedFQDN = (this._config.secure ? 'https://' : 'http://') + this._config.origin;
     this._coreParams = {};
 
     // create initial origins
     this.shiftStandardOrigin();
   }
 
   nextOrigin(): string {
-    // if a custom origin is supplied, use do not bother with shuffling subdomains
-    if (!this._providedFQDN.match(/ps\.pndsn\.com$/i)) {
-      return this._providedFQDN;
-    }
+    const protocol = this._config.secure ? 'https://' : 'http://';
 
-    let newSubDomain: string;
+    if (!this._config.isDomainShardingEnabled()) {
```

Comment:
I added a comment on the test for this below but maybe it should only enter here if both domain sharding enabled and has custom origin. Otherwise we will allow customers to accidentally disable sharding on internal ps.pndsn.com domains.